File: tournaments/addDigits/addDigits.cs
string addDigits(int a, int b, int n) {
    var number = a % b;
    var sb = new StringBuilder();

    sb.Append(a);

    for (int i = 0; n > i; ++i) {
        var nextDigit = -1;
        for (var digit = 9; 0 <= digit; --digit) {
            if (0 == (10 * number + digit) % b) {
                nextDigit = digit;
                break;
            }
        }
        if (-1 == nextDigit) {
            break;
        }
        sb.Append(nextDigit);
        number = (10 * number + nextDigit) % b;
    }

    return sb.ToString();
}

File: tournaments/addDigits/addDigits.cpp
std::string addDigits(int a, int b, int n) {

  int rem = a % b;

  std::string result = std::to_string(a);

  for (int i = 0; i < n; i++) {
    int best = -1;
    for (int digit = 9; digit >= 0; digit--) {
      if ((rem * 10 + digit) % b == 0) {
        best = digit;
        break;
      }
    }
    if (best == -1) {
      break;
    }
    result += '0' + best;
    rem = (rem * 10 + best) % b;
  }

  return result;
}
